pkg/proxy/routes.go

```diff
@@ -100,10 +100,19 @@ func removeToProxyRoutesIPv6() error {
 }
 
 var (
-	// Routing rule for traffic from proxy.
-	fromProxyRule = route.Rule{
+	// Routing rule for traffic from ingress proxy.
+	fromIngressProxyRule = route.Rule{
 		Priority: linux_defaults.RulePriorityFromProxyIngress,
-		Mark:     linux_defaults.MagicMarkIsProxy,
+		Mark:     linux_defaults.MagicMarkIngress,
+		Mask:     linux_defaults.MagicMarkHostMask,
+		Table:    linux_defaults.RouteTableFromProxy,
+		Protocol: linux_defaults.RTProto,
+	}
+
+	//nolint:unused // Routing rule for traffic from egress proxy.
+	fromEgressProxyRule = route.Rule{
+		Priority: linux_defaults.RulePriorityFromProxyIngress,
```

Comment:
Should this be the Ingress rule? From a quick glance I see we also have priority settings for egress rules